### ISSUE TITLE
BUG Ambiguous column "RecordID" when doing batch actions

### ISSUE THREAD

**Issue by tractorcow** [silverstripe/silverstripe-admin#579]
SilverStripe 4.x (all versions)

When doing batch actions on sitetree objects, queries can sometimes fail due to ambiguous RecordID column.

The bug is in CMSBatchActionHandler::getPages() method.

```php
$pages = Versioned::get_including_deleted($recordClass, array(
                '"RecordID" IN ('.DB::placeholders($ids).')' => $ids
            ));
```

Fix by

```php
$pages = Versioned::get_including_deleted($recordClass)->byIDs($ids);
```

And let versioned do the ID rewriting. :)

**Comment by tractorcow** [comment]
Fixed with https://github.com/silverstripe/silverstripe-admin/pull/581

**Comment by micahsheets** [comment]
Pull request #581 has not been merged into silverstripe-admin version 1.2.1. I am using SilverStripe-recipe-cms which requires a newer version of admin than 1.0 which this pull request was merged into. I would like to request that this fix be merged into silverstripe-admin 1.2.1.

**Comment by robbieaverill** [comment]
@micahsheets it needs to be merged up from 1.0. I'll do that now, and it should be released in the next patch releases for 1.0, 1.1 and 1.2
